Sources/MapboxCoreNavigation/CoreNavigationNavigator.swift

```diff
@@ -41,19 +41,24 @@ class Navigator {
         }
     }
     
-    let historyRecorder: HistoryRecorderHandle
+    private(set) var historyRecorder: HistoryRecorderHandle!
```

Comment:
Maybe it's better to move all these fields to a class/struct and recreate them all at once?